### PR TITLE
refactor(components): [steps] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by ryuhangyeong**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/steps/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  active: {
+    type: Number,
+    default: 0,
+  },
+  direction: {
+    type: String,
+    default: 'horizontal',
+    validator: (val: string): boolean =>
+      ['horizontal', 'vertical'].includes(val),
+  },
```
Comment (JeremyWuuuuu) [inline packages/components/steps/src/index.ts:20]:
```suggestion
  direction: {
    type: String,
    default: 'horizontal',
    values: ['horizontal', 'vertical'] as const,
  },
```

File: packages/components/steps/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  alignCenter: {
+    type: Boolean,
+    default: false,
```
Comment (JeremyWuuuuu) [inline packages/components/steps/src/index.ts:23]: We can remove `default: false` for `Boolean`

Simplified way:

```ts
{
  ...rest,
  alignCenter: Boolean,
}
```

File: packages/components/steps/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  simple: {
+    type: Boolean,
+    default: false,
+  },
```
Comment (JeremyWuuuuu) [inline packages/components/steps/src/index.ts:28]: Same here.

File: packages/components/steps/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  finishStatus: {
+    type: String,
+    default: 'finish',
+    validator: (val: string): boolean =>
+      ['wait', 'process', 'finish', 'error', 'success'].includes(val),
```
Comment (JeremyWuuuuu) [inline packages/components/steps/src/index.ts:33]:
```suggestion
      values: ['wait', 'process', 'finish', 'error', 'success'] as const,
```

File: packages/components/steps/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  processStatus: {
+    type: String,
+    default: 'process',
+    validator: (val: string): boolean =>
+      ['wait', 'process', 'finish', 'error', 'success'].includes(val),
```
Comment (JeremyWuuuuu) [inline packages/components/steps/src/index.ts:39]: Same here.

Maybe we can use a declaration for this:


```ts
const states = ['wait', 'process', 'finish', 'error', 'success'] as const
```

**Review by jw-foss — COMMENTED** [review]
@ryuhangyeong Left a few comments for you to consider 🏓 , if you need anything, you know how to find me ;)

File: packages/components/steps/src/index.ts (hunk 1 above)
Comment (ryuhangyeong) [reply packages/components/steps/src/index.ts:20]: Thank you! your Opinion!

File: packages/components/steps/src/index.ts (hunk 2 above)
Comment (ryuhangyeong) [reply packages/components/steps/src/index.ts:23]: Thank you! your Opinion!

File: packages/components/steps/src/index.ts (hunk 3 above)
Comment (ryuhangyeong) [reply packages/components/steps/src/index.ts:28]: Thank you! your Opinion!

File: packages/components/steps/src/index.ts (hunk 4 above)
Comment (ryuhangyeong) [reply packages/components/steps/src/index.ts:33]: Thank you! your Opinion!

File: packages/components/steps/src/index.ts (hunk 5 above)
Comment (ryuhangyeong) [reply packages/components/steps/src/index.ts:39]: Thank you! your Opinion!

Comment (ryuhangyeong) [reply packages/components/steps/src/index.ts]: @JeremyWuuuuu Hello~!  Thank you always for your good reviews. I have reflected your review!

**Comment by JeremyWuuuuu** [comment]
@ryuhangyeong You might need to do the same step in this PR as well.

**Comment by JeremyWuuuuu** [comment]
@ryuhangyeong You may rebase the `dev` now for getting `step` to avoid CI error.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**